airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/CdcPostgresSourceTest.java

```diff
@@ -383,4 +386,34 @@ protected void syncShouldHandlePurgedLogsGracefully() throws Exception {
     assertEquals(MODEL_RECORDS.size() + recordsToCreate + 1, recordsFromThirdBatch.size());
   }
 
+  @Test
+  void testReachedTargetPosition() {
```

Comment:
Not sure if this is feasible but - Is it possible to have a test that covers the e2e functionality?
e.g. logic that asserts that
- Heartbeats prevent a shutdown
- If the same LSN assc with a heartbeat is seen for a certain period of time, then the engine is shut down

Reply:
This is hard to achieve as heartbeats are generated inside debezium as opposed to change events coming from db.
With many classes that are package private.
I will file a tech debt ticket to try to do this.